session/create_producer.go

```diff
@@ -26,6 +26,7 @@ import (
 
 type createProducer struct {
 	ProposalID int
+	Config     json.RawMessage
```

Comment:
instead of depending heavily on json.RawMessage, a marker interface (with no methods) could be introduced and called ConsumerConfig

Reply:
For serialization it's not a problem as long as custom structure implements MarshalJSON interface. For deserialization - looks like the same problem as in communication - concrete message handler should return instance of message type for correct deserialization. Idea can be reused:
struct Message {
    field MarkerInterface
}

message.field = concreteStruct with MarshalJSON - communication serializes it correclty.
on deserialization side:
var message Message
message.field = concreteStruct
json.Unmarshal(&message). Should work, no?